scipy/stats/_discrete_distns.py

```diff
@@ -514,6 +526,124 @@ def _logcdf(self, k, M, n, N):
+    definitions used here.
+
+    The probability mass function is defined as,
+
+    .. math:: f(k; N, K, r) = \frac{{{k+r-1}\choose{k}}{{N-r-k}\choose{K-k}}}
+                                   {{N \choose K}}
+
+    for :math:`k \in [0, K]` and the binomial co-efficient is:
+
+    .. math:: \binom{n}{k} \equiv \frac{n!}{k! (n - k)!}.
+
+    %(after_notes)s
+
+    Examples
+    --------
+    >>> from scipy.stats import nhypergeom
+    >>> import matplotlib.pyplot as plt
+    >>> import numpy as np
+
+    Suppose we have a collection of 20 animals, of which 7 are dogs.
+    Then if we want to know the probability of finding a given number
+    of dogs (successes) in a sample with exactly 12 animals that
+    aren't dogs (failures), we can initialize a frozen distribution
+    and plot the probability mass function:
+
+    >>> N, K, r = [20, 7, 12]
+    >>> rv = nhypergeom(N, K, r)
+    >>> x = np.arange(0, K+2)
+    >>> pmf_dogs = rv.pmf(x)
+
+    >>> fig = plt.figure()
+    >>> ax = fig.add_subplot(111)
+    >>> ax.plot(x, pmf_dogs, 'bo')
+    >>> ax.vlines(x, 0, pmf_dogs, lw=2)
+    >>> ax.set_xlabel('# of dogs in our group with given 12 failures')
+    >>> ax.set_ylabel('nhypergeom PMF')
+    >>> plt.show()
+
+    Instead of using a frozen distribution we can also use `nhypergeom`
+    methods directly.  To for example obtain the probability mass
+    function, use:
+
+    >>> prb = nhypergeom.pmf(x, N, K, r)
+
+    And to generate random numbers:
+
+    >>> R = nhypergeom.rvs(N, K, r, size=10)
+
+    See Also
+    --------
+    hypergeom, binom, nbinom
+
+    References
+    ----------
+    .. [1] Negative Hypergeometric Distribution on Wikipedia
+           https://en.wikipedia.org/wiki/Negative_hypergeometric_distribution
+
+    .. [2] Negative Hypergeometric Distribution from
+           http://www.math.wm.edu/~leemis/chart/UDR/PDFs/Negativehypergeometric.pdf
+
+    """
+    def _get_support(self, N, K, r):
+        return 0, K
+
+    def _argcheck(self, N, K, r):
+        cond = (N >= 0)
+        cond &= (K >= 0) & (K <= N)
+        cond &= (r >= 0) & (r <= N-K)
+        return cond
+
+    def _logpmf(self, k, N, K, r):
+        result = (-betaln(k+1, r) + betaln(k+r, 1) -
+                  betaln(K-k+1, N-r-K+1) + betaln(N-r-k+1, 1) +
+                  betaln(K+1, N-K+1) - betaln(N+1, 1))
+        return result
+
+    def _pmf(self, k, N, K, r):
+        # same as the following but numerically more precise
+        # return comb(k+r-1, k) * comb(N-r-k, K-k) / comb(N, K)
+        return exp(self._logpmf(k, N, K, r))
+
+    def _stats(self, N, K, r):
+        # Promote the datatype to at least float
+        # mu = rK / (N-K+1)
+        N, K, r = 1.*N, 1.*K, 1.*r
+        mu = r*K / (N-K+1)
+
+        var = r*(N+1)*K / ((N-K+1)*(N-K+2))
+        var *= 1 - r / (N-K+1)
+
+        # The skew and kurtosis are mathematically
+        # intractable so return np.nan. See [2].
```

Comment:
Reference added and added here too!

Reply:
Does this make skew and kurtosis fall back on the superclass method? (I would need to build this branch again to check it out, and I'm working on another.)

Reply:
`stats` is defined in `rv_generic` and looking at the method, it doesn't seem to do anything when skew and kurtosis are nan.

Reply:
There is generic code to calculate any moment from the pmf when a method-specific definition is not available. I think it is activated by returning `None`. I changed these to `None` and I get numbers, but please trace where they're coming from and check that they're correct. I would also suggest trying out all the other methods available for the distribution. I don't think anything should return `nan`. Everything should work. It might be slow - but it should work.

Reply:
Oh, yes! You are right. It only returns `nan`s if the input is invalid. So, I think changing the `nan` to `None` should do it.